### ISSUE TITLE
KeyboardSpacer does not work properly with react navigation tab bar

### ISSUE THREAD

**Issue by mikechambers610** [Andr3wHur5t/react-native-keyboard-spacer#72]
When using the keyboard spacer it will push the textinput up an additional amount (additional amount = height of the tab bar) for both ios and android.

Works great if you aren't using a tab bar.... but most cases the developer likes to use the tab bars.

**Comment by arthurCalv** [comment]
Yeah having the same issue with a grey bar above the keyboard when using a tab bar
![screen shot 2018-06-14 at 17 33 44](https://user-images.githubusercontent.com/37707381/41425709-739dbfe0-6ff9-11e8-860f-334e6155f613.png)

**Comment by pulpdood** [comment]
I added topSpacing={-50} and it seems to fix the issue:

`<KeyboardSpacer topSpacing={-50}/>`

**Comment by mikechambers610** [comment]
Thanks pulpdood I will give it a shot and see how it works.

**Comment by hieunc229** [comment]
I experienced the issue while applying into an 'absolute' or 'fixed' view. I could be caused by any issue, though I found out that while using 'absolute' or 'fixed' view, the calculation would be missed ([similar to this described issue](https://github.com/facebook/react-native/issues/10743)). 

I came up with a different method that supports 'absolute', 'fixed' view and publish it on [react-native-spacer](https://ambistudio.github.io/react-native-spacer/).

**Comment by dextermb** [comment]
@pulpdood while this works on some devices, it does not seem to work on all. For example I still get bars on Pixel XL 2 and iPhone X when adding `-55` on `topSpacing`

**Comment by ESA2012** [comment]
Yet another solution:
```
import { getBottomSpace } from 'react-native-iphone-x-helper';
...
<KeyboardSpacer topSpacing={-getBottomSpace()} />
```

**Comment by CyxouD** [comment]
thanks for direction @ESA2012. For the case of iPhoneX + tab bar. I use:
```
import { getBottomSpace } from 'react-native-iphone-x-helper';
...
<KeyboardSpacer topSpacing={-(TAB_BAR_HEIGHT + getBottomSpace())} />
```
It looks like it is exact this value needed

My config:
```
"react-native": "0.59.10",
"react-navigation": "^3.0.5"
```

**Comment by alansp** [comment]
I'm using this for iPhone only (don't have issues on Android), and this combo solved all my problems...

`import { useBottomTabBarHeight } from '@react-navigation/bottom-tabs';`

`<KeyboardSpacer topSpacing={-useBottomTabBarHeight()} />`
